datastore/user_keys.go

```diff
@@ -32,10 +34,40 @@ func (DBUserKey) TableName() string {
 
 // StoreUserKey saves a user key to the database
 func (d *Datastore) StoreUserKey(key *DBUserKey) error {
-	// Count existing keys for this service/account pair, excluding the current key_name
+	return d.DB.Transaction(func(tx *gorm.DB) error {
```

Comment:
I suspect that we may be able to avoid having separate update + create functions within a transaction by performing a [single insert that updates on conflict](https://gorm.io/gen/create.html#Upsert-On-Conflict).

Reply:
Done.